conda/petsc/build.sh

```diff
@@ -1,119 +1,164 @@
+#  cd $BUILD_PREFIX/lib
+#  rm -f libnvToolsExt.so
+#  ln -s ../targets/x86_64-linux/lib/libnvToolsExt.so.1.0.0 libnvToolsExt.so
+#  cd -
+#
+#  CXXFLAGS+=" -I${PREFIX}/nsight-compute-2024.1.0/host/target-linux-x64/nvtx/include/nvtx3"
+#  CFLAGS+=" -I${PREFIX}/nsight-compute-2024.1.0/host/target-linux-x64/nvtx/include/nvtx3"
+#  ADDITIONAL_ARGS+=" --download-slate=1 --with-cuda=1 --with-cudac=${PREFIX}/bin/nvcc --with-cuda-dir=${PREFIX}/targets/x86_64-linux --CUDAFLAGS=-I${PREFIX}/targets/x86_64-linux/include"
+#fi
 
-source $PETSC_DIR/configure_petsc.sh
+source $SRC_DIR/configure_petsc.sh
 configure_petsc \
-    --COPTFLAGS=-O3 \
-    --CXXOPTFLAGS=-O3 \
-    --FOPTFLAGS=-O3 \
-    --with-x=0 \
-    --with-ssl=0 \
-    --with-mpi-dir=$PREFIX \
-    AR="$AR" \
-    RANLIB="$RANLIB" \
-    CFLAGS="$CFLAGS" \
-    CXXFLAGS="$CXXFLAGS" \
-    CPPFLAGS="$CPPFLAGS" \
-    FFLAGS="$FFLAGS" \
-    FCFLAGS="$FCFLAGS" \
-    LDFLAGS="$LDFLAGS" \
-    --prefix=$PREFIX || (cat configure.log && exit 1)
-
-# Verify that gcc_ext isn't linked
-for f in $PETSC_ARCH/lib/petsc/conf/petscvariables $PETSC_ARCH/lib/pkgconfig/PETSc.pc; do
-  if grep gcc_ext $f; then
-    echo "gcc_ext found in $f"
-    exit 1
-  fi
-done
+      --COPTFLAGS=-O3 \
+      --CXXOPTFLAGS=-O3 \
+      --FOPTFLAGS=-O3 \
+      --with-x=0 \
+      --with-ssl=0 \
+      --with-hdf5-dir=$HDF5_DIR \
+      --with-mpi-dir=$PREFIX \
+      AR="$AR" \
+      RANLIB="$RANLIB" \
+      CFLAGS="$CFLAGS" \
+      CXXFLAGS="$CXXFLAGS" \
+      CPPFLAGS="$CPPFLAGS" \
+      FFLAGS="$FFLAGS" \
+      FCFLAGS="$FFLAGS" \
+      LDFLAGS="$LDFLAGS" \
+      ${ADDITIONAL_ARGS} \
+      --prefix=$PREFIX/petsc || (tail -400 configure.log && exit 1)
 
+## Verify that gcc_ext isn't linked
+#for f in $PETSC_ARCH/lib/petsc/conf/petscvariables $PETSC_ARCH/lib/pkgconfig/PETSc.pc; do
+#  if grep gcc_ext $f; then
+#    echo "gcc_ext found in $f"
+#    exit 1
+#  fi
+#done
+#
 sedinplace() {
   if [[ $(uname) == Darwin ]]; then
     sed -i "" "$@"
   else
     sed -i"" "$@"
   fi
 }
+#
+## Remove abspath of ${BUILD_PREFIX}/bin/python
+#sedinplace "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/include/petscconf.h
+#sedinplace "s%${BUILD_PREFIX}/bin/python%python%g" $PETSC_ARCH/lib/petsc/conf/petscvariables
+#sedinplace "s%${BUILD_PREFIX}/bin/python%/usr/bin/env python%g" $PETSC_ARCH/lib/petsc/conf/reconfigure-arch-conda-c-opt.py
+#
+## Replace abspath of ${PETSC_DIR} and ${BUILD_PREFIX} with ${PREFIX}
+#for path in $PETSC_DIR $BUILD_PREFIX; do
+#    for f in $(grep -l "${path}" $PETSC_ARCH/include/petsc*.h); do
+#        echo "Fixing ${path} in $f"
+#        sedinplace s%$path%\${PREFIX}/moose-petsc/%g $f
+#    done
+#done
+#
+#make
+## damn... again I have to disable this. (openmpi strange missing libraries error)
+#if [[ $mpi == "mpich" ]] && [[ $(uname) == Linux ]]; then
+#  make check
+#fi
+make PETSC_DIR=$SRC_DIR PETSC_ARCH=$PETSC_ARCH all
+make PETSC_DIR=$SRC_DIR PETSC_ARCH=$PETSC_ARCH install
```

Comment:
`make check` bites the dust ☠️ 

Nice touch in robustify-ing the `make` lines with the env variables.

Reply:
I'll put it back :) Maybe with the whole
```bash
FI_PROVIDER=tcp
```
it will now work!